torchvision/models/detection/backbone_utils.py

```diff
@@ -152,7 +154,10 @@ def _validate_trainable_layers(
     # by default freeze first blocks
     if trainable_backbone_layers is None:
         trainable_backbone_layers = default_value
-    assert 0 <= trainable_backbone_layers <= max_value
+        if trainable_backbone_layers not in range(0, max_value + 1):
```

Comment:
Again range() would be bit slower. This is faster.

```suggestion
        if trainable_backbone_layers < 0 or trainable_backbone_layers > max_value:
```

Reply:
yes, replacing range generator with simple logic